pkgs/build-support/setup-hooks/writable-tmpdir-as-home.sh

```diff
@@ -0,0 +1,15 @@
+# shellcheck shell=bash
+# This setup hook set the HOME environment variable to a writable directory.
+
+export HOME
+
+writableTmpDirAsHome () {
+    if [ ! -w "$HOME" ]; then
+        HOME="$NIX_BUILD_TOP/.home"
```

Comment:
Why is that a hidden directory and not just home?

Reply:
It has been suggested to name it as such, I don't really have any preference.